<launch>
  <arg name="interface_name" default="wlan0"/>
  <arg name="rssi_sampling_rate" default="1" />
  <arg name="other_sampling_rate" default="1" />
  <arg name="node_name" default="client" />

  <node pkg="network_analysis" type="wireless_quality.py" name="wireless_quality" output="screen">
    <param name="INTERFACE_NAME" value="$(arg interface_name)"/>
    <param name="update_rate_wireless_quality" value="$(arg rssi_sampling_rate)"/>
    <remap from="/network_analysis/wireless_quality" to="$(arg node_name)/network_analysis/wireless_quality" />
  </node>
  <node pkg="network_analysis" type="link_utilization.py" name="link_utilization" output="screen">
    <param name="INTERFACE_NAME" value="$(arg interface_name)"/>
    <param name="update_rate_link_utilization" value="$(arg other_sampling_rate)" />
    <remap from="/network_analysis/link_utilization" to="$(arg node_name)/network_analysis/link_utilization" />
  </node>
  <node pkg="network_analysis" type="network_delay" name="network_delay" output="screen">
    <param name="update_rate_network_delay" value="$(arg other_sampling_rate)" />    
    <param name="timeout_network_delay" value="2.0" />    
    <remap from="/network_analysis/network_delay" to="$(arg node_name)/network_analysis/network_delay" />
  </node>
  <node pkg="network_analysis" type="network_errors.py" name="network_errors" output="screen">
    <param name="INTERFACE_NAME" value="$(arg interface_name)"/>
    <param name="update_rate_network_errors" value="$(arg other_sampling_rate)" />    
    <remap from="/network_analysis/network_errors" to="$(arg node_name)/network_analysis/network_errors" />    
  </node>
</launch>
